app/src/renderer/vuex/modules/blockchain.js

```diff
@@ -30,21 +36,33 @@ export default ({ commit, node }) => {
       dispatch("subscribeToBlocks")
     },
     async getBlock({ state, commit, dispatch }, height) {
-      state.blockLoading = true
-      state.blockHeight = height
-      return Promise.all([
-        dispatch("queryBlock", height).then(block => commit("setBlock", block)),
-        dispatch("queryBlockInfo", height).then(blockMetaInfo =>
-          commit("setBlockMetaInfo", blockMetaInfo)
-        )
-      ]).then(
-        () => {
-          state.blockLoading = false
-        },
-        () => {
-          state.blockLoading = false
-        }
-      )
+      return new Promise((resolve, reject) => {
+        state.blockLoading = true
+        state.blockHeight = height
+        return Promise.all([
+          dispatch("queryBlock", height).then(block =>
+            commit("setBlock", block)
+          ),
+          dispatch("queryBlockInfo", height).then(blockMetaInfo =>
+            commit("setBlockMetaInfo", blockMetaInfo)
+          )
+        ])
+          .then(() => {
+            state.blockLoading = false
+            dispatch("queryTxInfo", height)
```

Comment:
If you return this, I think you save yourself the inner .catch

Reply:
refactored all the promises to be more concise